src/main/scala/com/cognite/spark/datasource/DefaultSource.scala

```diff
@@ -24,6 +24,8 @@ case class RelationConfig(
 object OnConflict extends Enumeration {
   type Mode = Value
   val ABORT, UPDATE, UPSERT, DELETE = Value
+
+  def withNameOpt(s: String): Option[Value] = values.find(_.toString.toLowerCase == s.toLowerCase())
```

Comment:
Can't we use
https://www.scala-lang.org/api/current/scala/Enumeration.html#withName(s:String):Enumeration.this.Value?

Or is that method not case insensitive?

Reply:
The problem with that method is that it throws the exception referenced in the related issue. The new function returns an Option and allows us to throw a more specific error message.

Reply:
Oh, I see then it makes sense. And it's also nicer than relying on exception for code flow, as when doing `Try(OnConflict.withName(parameters.getOrElse("onconflict", "ABORT").toUpperCase())).toOption`.